mcp/src/main/java/io/modelcontextprotocol/server/McpAsyncServer.java

```diff
@@ -680,12 +694,23 @@ public Mono<Void> loggingNotification(LoggingMessageNotification loggingMessageN
 			return this.mcpTransportProvider.notifyClients(McpSchema.METHOD_NOTIFICATION_MESSAGE, params);
 		}
 
-		private McpServerSession.RequestHandler<Void> setLoggerRequestHandler() {
+		private McpServerSession.RequestHandler<Object> setLoggerRequestHandler() {
 			return (exchange, params) -> {
-				this.minLoggingLevel = objectMapper.convertValue(params, new TypeReference<LoggingLevel>() {
-				});
+				return Mono.defer(() -> {
 
-				return Mono.empty();
+					SetLevelRequest newMinLoggingLevel = objectMapper.convertValue(params,
+							new TypeReference<SetLevelRequest>() {
+							});
+
+					exchange.setMinLoggingLevel(newMinLoggingLevel.level());
```

Comment:
I didn't find an actual implementation of this in the python reference SDK. Have you checked others? Is there a consensus here? I am wondering if it's the right way to implement it, as at least in the Python SDK the handler delegates to a user-provided function to handle it.

Reply:
Ok thanks! Having consulted other impls, I think it's correct for us to store the state in the exchange and not expose this to the users at all.